File: VB/第二套/11900-1060306/Program.vb
Imports System.Windows.Forms

Module Program
    <STAThread()>
    Sub Main()
        Application.EnableVisualStyles()
        Application.SetCompatibleTextRenderingDefault(False)

        Dim view As New Views.MainForm()
        Dim validator As New Services.IdCardValidatorService()
        Dim dataService As New Services.DataService()
        
        Dim presenter As New Presenters.MainPresenter(view, validator, dataService)
        
        Application.Run(view)
    End Sub
End Module

File: C#/第二套/11900-1060306/Program.cs
using Microsoft.Extensions.DependencyInjection;
using IdCardChecker.Views;
using IdCardChecker.Presenters;
using IdCardChecker.Services;

namespace IdCardChecker;

static class Program
{
    [STAThread]
    static void Main()
    {
        ApplicationConfiguration.Initialize();

        var services = new ServiceCollection();

        // Register Services
        services.AddSingleton<IIdCardValidatorService, IdCardValidatorService>();
        services.AddSingleton<IDataService, DataService>();

        // Register Views
        services.AddSingleton<IMainView, MainForm>();

        // Register Presenters
        services.AddSingleton<MainPresenter>();

        using var serviceProvider = services.BuildServiceProvider();

        // Resolve Presenter to wire up events
        var presenter = serviceProvider.GetRequiredService<MainPresenter>();
        var mainForm = (Form)serviceProvider.GetRequiredService<IMainView>();

        Application.Run(mainForm);
    }
}
